recipes/dd-handler.rb

```diff
@@ -32,10 +32,7 @@
 chef_gem 'chef-handler-datadog' do # ~FC009
   action :install
   version node['datadog']['chef_handler_version']
-  # Chef 12 introduced `compile_time` - remove when Chef 11 is EOL.
-  compile_time true if respond_to?(:compile_time)
 end
-require 'chef/handler/datadog'
```

Comment:
Took some time to test this, and unfortunately moving the `require` doesn't change its evaluation time, it's still evaluated at compile time.

This commit still works in Chef 12 because not specifying `compile_time` on `chef_gem` still makes the install of `chef-handler-datadog` happen at compile time (see, in the link you provided: `The default behavior has not changed, but every `chef_gem` resource will now emit out a warning`). The default will actually change to `compile_time false` in Chef 13.

So we should keep the `compile_time true` line until we can get rid of the `require` altogether (and I _think_ we'll be able to do that once we bump the requirement on `chef_handler` to `~> 1.3`, see related https://github.com/DataDog/chef-datadog/issues/338#issuecomment-238295004 ). Let's keep this discussion in a separate issue/PR though.